app/model/Conditions.scala

```diff
@@ -58,22 +59,57 @@ object HasAnyBusinessEnrolment extends HasAnyBusinessEnrolment {
 
 object SAReturnAvailable extends Condition {
   override def isTrue(authContext: AuthContext, ruleContext: RuleContext)(implicit request: Request[AnyContent], hc: HeaderCarrier) =
-    ruleContext.lastSaReturn.map(_ => true).recover { case _ => false}
+    ruleContext.lastSaReturn.map(_ => true).recover { case _ => false }
 
   override val auditType = Some(SA_RETURN_AVAILABLE)
 }
 
-trait HasSelfAssessmentEnrolments extends Condition {
-  def selfAssessmentEnrolments: Set[String]
+sealed trait EnrolmentCategory {
+  def enrolmentCategoryName: String
+
+  def enrolments: Set[String]
+}
+
+trait EnrolmentCategoryFromConf extends EnrolmentCategory {
+  def enrolments = Play.configuration.getStringList(enrolmentCategoryName).map(_.toSet).getOrElse(Set.empty[String])
+}
+
+case class ConfiguredEnrolmentCategory(enrolmentCategoryName: String) extends EnrolmentCategoryFromConf
+
+object SA extends ConfiguredEnrolmentCategory("self-assessment-enrolments")
+
+object VAT extends ConfiguredEnrolmentCategory("vat-enrolments")
+
+trait HasEnrolmentsCondition extends Condition {
+
+  protected def strict: Boolean
+
+  def enrolmentCategories: Set[EnrolmentCategory]
+
+  private def validateAllEnrolmentCategoriesExist(enrolments: Set[String]) = enrolmentCategories.forall(_.enrolments.intersect(enrolments).nonEmpty)
+
+  private def validateExtraEnrolments(enrolments: Set[String]) = !strict || (strict && enrolments.diff(enrolmentCategories.flatMap(_.enrolments)).isEmpty)
 
   override def isTrue(authContext: AuthContext, ruleContext: RuleContext)(implicit request: Request[AnyContent], hc: HeaderCarrier) =
-    ruleContext.activeEnrolments.map(_.intersect(selfAssessmentEnrolments).nonEmpty)
+    ruleContext.activeEnrolments.map(userEnrolments => validateAllEnrolmentCategoriesExist(userEnrolments) && validateExtraEnrolments(userEnrolments))
+}
+
+case class HasOnlyEnrolments(enrolmentCategories: Set[EnrolmentCategory]) extends HasEnrolmentsCondition {
+  protected val strict = true
+  override val auditType = Some(HAS_ONLY_ENROLMENTS(enrolmentCategories))
+}
+
+case class HasEnrolments(enrolmentCategories: Set[EnrolmentCategory]) extends HasEnrolmentsCondition {
+  protected val strict = false
+  override val auditType = Some(HAS_ENROLMENTS(enrolmentCategories))
+}
 
-  override val auditType = Some(HAS_SA_ENROLMENTS)
+object HasOnlyEnrolments {
+  def apply(enrolmentCategories: EnrolmentCategory*): HasOnlyEnrolments = HasOnlyEnrolments(enrolmentCategories.toSet)
 }
 
-object HasSelfAssessmentEnrolments extends HasSelfAssessmentEnrolments {
-  override lazy val selfAssessmentEnrolments = Play.configuration.getString("self-assessment-enrolments").getOrElse("").split(",").map(_.trim).toSet[String]
+object HasEnrolments {
+  def apply(enrolmentCategories: EnrolmentCategory*): HasEnrolments = HasEnrolments(enrolmentCategories.toSet)
```

Comment:
usually I don't like varargs, but it is good here because it makes the TarRules easy to read